templates/demo-store/app/components/AddToCartButton.tsx

```diff
@@ -31,26 +39,73 @@ export function AddToCartButton({
       }}
       action={CartForm.ACTIONS.LinesAdd}
     >
-      {(fetcher: FetcherWithComponents<any>) => (
-        <>
-          <input
-            type="hidden"
-            name="analytics"
-            value={JSON.stringify(analytics)}
-          />
-          <Button
-            as="button"
-            type="submit"
-            width={width}
-            variant={variant}
-            className={className}
-            disabled={disabled ?? fetcher.state !== 'idle'}
-            {...props}
-          >
-            {children}
-          </Button>
-        </>
-      )}
+      {(fetcher: FetcherWithComponents<any>) => {
+        return (
+          <AddToCartAnalytics fetcher={fetcher}>
+            <input
+              type="hidden"
+              name="analytics"
+              value={JSON.stringify(analytics)}
+            />
+            <Button
+              as="button"
+              type="submit"
+              width={width}
+              variant={variant}
+              className={className}
+              disabled={disabled ?? fetcher.state !== 'idle'}
+              {...props}
+            >
+              {children}
+            </Button>
+          </AddToCartAnalytics>
+        );
+      }}
     </CartForm>
   );
 }
+
+function AddToCartAnalytics({
+  fetcher,
+  children,
+}: {
+  fetcher: FetcherWithComponents<any>;
+  children: React.ReactNode;
+}): JSX.Element {
+  const fetcherData = fetcher.data;
+  const formData = fetcher.formData;
+  const pageAnalytics = usePageAnalytics({hasUserConsent: true});
+
+  useEffect(() => {
+    if (formData) {
+      const cartData: Record<string, unknown> = {};
+      const cartInputs = CartForm.getFormInput(formData);
+
+      try {
+        if (cartInputs.inputs.analytics) {
+          const dataInForm: unknown = JSON.parse(
+            String(cartInputs.inputs.analytics),
+          );
+          Object.assign(cartData, dataInForm);
+        }
+      } catch {
+        // do nothing
+      }
+
+      if (Object.keys(cartData).length && fetcherData) {
+        const addToCartPayload: ShopifyAddToCartPayload = {
+          ...getClientBrowserParameters(),
+          ...pageAnalytics,
+          ...cartData,
+          cartId: fetcherData.cart.id,
+        };
+
+        sendShopifyAnalytics({
```

Comment:
Any concern with React 18 strict mode running `useEffect` twice?

Reply:
it runs twice but because only 1 instance will have `cartData` and `fetcherData`, this is fine